src/client/common/application/encryptedStorage.ts

```diff
@@ -38,10 +41,21 @@ const keytar = getNodeModule<KeyTar>('keytar');
 export class EncryptedStorage implements IEncryptedStorage {
     constructor(
         @inject(IApplicationEnvironment) private readonly appEnv: IApplicationEnvironment,
-        @inject(IAuthenticationService) private readonly authenService: IAuthenticationService
+        @inject(IAuthenticationService) private readonly authenService: IAuthenticationService,
+        @inject(IExtensionContext) private readonly extensionContext: IExtensionContext
     ) {}
 
+    private readonly testingState = new Map<string, string>();
+
     public async store(service: string, key: string, value: string | undefined): Promise<void> {
+        traceInfo(`Start Setup.F1`);
+        // On CI we don't need to use keytar for testing (else it hangs).
+        if (IS_REMOTE_NATIVE_TEST && this.extensionContext.extensionMode !== ExtensionMode.Production) {
```

Comment:
Found that this part hangs on CI, i think this is because we don't have a proper unix user on CI &/or keytar isn't working for some other reason.
Either way, if its not production & its remote, then i've disabled this.
Should probably disable completely on CI & not just remote tests

Let me know what you (reviewer) think.
I personally think we should use internal state when on CI

@rchiodo @IanMatthewHuff @DavidKutu @joyceerhl /cc

Reply:
Seems alright in this case. It's getting tested in other scenarios.